src/main/webapp/script.js

```diff
@@ -0,0 +1,100 @@
+var zoomedIn = false;
+var aqLayerOn = true;
+var routeVisible = false;
+
+function changeBackgroundImage(imagePath) {
+    document.body.style.backgroundImage = "url('" + imagePath + "')";
+}
+
+function getImagePath() {
+    path = "";
+    if (zoomedIn) {
+        path = "images/wagga_";
+    } else {
+        path = "images/nsw_";
+    }
+
+    if (aqLayerOn) {
+        path += "gradient";
+    } else {
+        path += "plain";
+    }
+
+    if (routeVisible) {
+        path += "_route.png";
+    } else {
+        path += ".png";
+    }
+
+    return path;
+}
+
+function toggleAQLayer() {
+    checkbox = document.getElementById("aq-layer-toggle");
+    
+    aqLayerOn = checkbox.checked;
+
+    changeBackgroundImage(getImagePath());
+}
+
+function hideRoutes() {
+    routeVisible = false;
```

Comment:
Again; the CSS can be used as the active state, queries and toggled; rather than managed in the script.

Reply:
I'm not sure what you mean by this

Reply:
Rather than using the global scope `routeVisible` variable; you can read the value from `document.getElementById("close-directions").style.visibility`, and use that instead.